craft_store/publisher/_publishergw.py

```diff
@@ -170,8 +172,8 @@ def list_registered_names(
             params={"include-collaborations": include_collaborations},
         )
         self._check_error(response)
-        results = self._check_keys(response, expected_keys={"results"})["results"]
-        return [RegisteredName.unmarshal(item) for item in results]
+        response_data = response.json()
```

Comment:
We still do need `check_keys` here since the resulting JSON doesn't directly get validated.